### PR TITLE
chore: Add kfp-tekton integration tests and manifests

### PR DESCRIPTION
**Pull request by rimolive**
**Description of your changes:**
Solves #10489

Adds the kfp-tekton integration tests and corresponding manifests

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by rimolive** [comment]
/hold

Wait for tests to pass before ask for review/approval

**Review by Tomcli — CHANGES_REQUESTED** [review]
few comments on the registry name.

File: manifests/kustomize/third-party/tekton-custom-task/driver-controller/200-serviceaccount.yaml
Comment (Tomcli) [inline manifests/kustomize/third-party/tekton-custom-task/driver-controller/200-serviceaccount.yaml:1]: FYI everything under the `/manifests/kustomize/third-party/tekton-custom-task/driver-controller` was for the old v2 implementation. We moved all the driver logic to the Tekton custom kfptask to handle driver and task creation logic in the same Tekton controller.

File: manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml
Comment (rimolive) [reply manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml:17]: Only this image or all of them?

File: scripts/deploy/github/re-tag.sh
Comment (rimolive) [reply scripts/deploy/github/re-tag.sh:22]: Where this script is used? Maybe it's safe to remove?

File: manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml
Comment (Tomcli) [reply manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml:17]: All the images under `manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml` and ``manifests/kustomize/env/platform-agnostic-tekton/kustomization.yaml`

File: scripts/deploy/github/re-tag.sh
Comment (Tomcli) [reply scripts/deploy/github/re-tag.sh:22]: This script is to retag the local built docker.io images to the gcr.io images. So the local test can use the same manifests but only use the images in the local registry.

Comment (Tomcli) [reply scripts/deploy/github/re-tag.sh:22]: The renamed images in this script will not push to external, so we don't need to worry about push credentials etc.

File: .github/workflows/backend.yml
Comment (Tomcli) [inline .github/workflows/backend.yml:40]: I found the deployment is missing some manifests file. It probably due to the recent update to the new kustomize 5.3 version. To fix that, we need to upgrade the kubectl version to 1.32+ or convert all the kubectl -k commands to `kustomize`

Comment (rimolive) [reply .github/workflows/backend.yml]: I changed to 1.29.4 because of the k8s 1.29 support we're planning for Kubeflow 1.9. I'll check the kubectl -k change to kustomize.

**Comment by rimolive** [comment]
Test are failling because some resources don't have namespace explicitly set. Some examples:
* [MySQL](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/third-party/mysql/base/mysql-deployment.yaml#L4)
* [ml-metadata Envoy Proxy](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/base/metadata/base/metadata-envoy-service.yaml#L6)
* [ml-metadata gRPC Server](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/base/metadata/base/metadata-grpc-deployment.yaml#L4)

I tried to find a way to set in the kustomization.yaml level to set namespace in the objects that don't have namespace set but no luck. That being said, I'd like to ask @Tomcli and @chensun :
* Is it ok if I set namespace in these Deployments?
* If not, do you suggest another way to make them deploy in kubeflow namespace without explicitly setting in these objects?

**Review by Tomcli — COMMENTED** [review]
few comments on fixing the old kustomize code to kustomize 5.2.1

**Review by Tomcli — COMMENTED** [review]
Also, I think you didn't copy some role permission from the persistent agent, so the end to end test wasn't able to update the run status.

https://github.com/kubeflow/kfp-tekton/blob/v2-integration/manifests/kustomize/base/pipeline/ml-pipeline-persistenceagent-role.yaml#L29-L45

File: manifests/kustomize/third-party/tekton/upstream/manifests/base/tektoncd-install/tekton-release.yaml
Comment (rimolive) [inline manifests/kustomize/third-party/tekton/upstream/manifests/base/tektoncd-install/tekton-release.yaml:1704]: @Tomcli One quick thing: I had to change the value of that feature flag because otherwise kfptask and exit-handler webhooks will keep failling. If you have more details about that flag and if it's ok to keep it set that way, I really appreciate.

**Comment by rimolive** [comment]
> Also, I think you didn't copy some role permission from the persistent agent, so the end to end test wasn't able to update the run status.
> 
> https://github.com/kubeflow/kfp-tekton/blob/v2-integration/manifests/kustomize/base/pipeline/ml-pipeline-persistenceagent-role.yaml#L29-L45

Thank you for that! I'm still trying to figure you why the flip coin pipeline never finishes. At least with that change I can see the pipeline status of RUNNING, but it stays in that status until failure.

File: manifests/kustomize/third-party/tekton/upstream/manifests/base/tektoncd-install/tekton-release.yaml
Comment (Tomcli) [reply manifests/kustomize/third-party/tekton/upstream/manifests/base/tektoncd-install/tekton-release.yaml:1704]: spire is an extra server for signing Tekton results and artifacts. It's not really a feature we need for KFP and enable it will reduce the Tekton performance.

Do you have the error message for the kfptask and exit-handler webhooks? We probably just need to update the config for these two webhooks to support k8s 1.29.

File: .github/workflows/backend.yml
Comment (Tomcli) [inline .github/workflows/backend.yml:41]: I'm not sure kind 1.17 has the bug fixes for the k8s 1.29 cluster deployment. Maybe we either upgrade kind to 0.22 which has k8s 1.29 support or just run `node_image: kindest/node:v1.26.3` to minimize the deployment variable changes.
```suggestion
          version: v0.22.0
```

File: manifests/kustomize/third-party/tekton/upstream/manifests/base/tektoncd-install/tekton-release.yaml
Comment (rimolive) [reply manifests/kustomize/third-party/tekton/upstream/manifests/base/tektoncd-install/tekton-release.yaml]: Nevermind, I reverted that change and deployment looks good.

**Comment by rimolive** [comment]
Just a checkpoint comment: Tests are passing, but I still need to address Tommy's comment on Kustomize 5.2.1 support and the re-tag script.

**Comment by rimolive** [comment]
/unhold
cc @Tomcli @chensun

**Review by DharmitD — COMMENTED** [review]
Skimmed through the integration test results in CI and verified they work as expected.

/lgtm

**Review by chensun — APPROVED** [review]

**Comment by rimolive** [comment]
@chensun Looks like e2e test did not finish well, but unrelated to the PR. Should we try run again?

**Pull request merged**
